### PR TITLE
Adding Version Endpoints to all apis

### PR DESCRIPTION
**Pull request by hattan**
## Purpose
This PR introduces new endpoints for all apis. The endpoint is /api/version/<api> 

This is done via a docker environment variable that is assigned during the docker build phase. This can be set via the command line by adding --build-arg to docker build.

The purpose of this is to make it easy and provide a standard way for teams to see updates to their api. Rather than having to mess with the code for each api (and change healthchecks) , teams can easily add --build-arg to their pipeline. 

For example:

``` 
docker build -t api-poi:$(Build.BuildId) --build-arg build_version=$(BuildBuildId)
```

By doing this during an automated build the deployed api will always show a unique version number in it's version endpoint. This makes it much easier for folks to see changes without needing to update the api code itself.

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code
*  for each api:
(poi example)
```
  cd apis/poi/web
  docker build -t api-poi:test123 --build-arg build_version=test123 .
  docker run -p 8080:8080 api-poi
  curl localhost:8080/api/version/poi

### CODE REVIEW

File: apis/trips/tripsgo/versionService.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	"os"
+)
+
+func versionGet(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "text/plain; charset=UTF-8")
+	w.WriteHeader(http.StatusOK)
+
+	version := os.Getenv("APP_VERSION")
+	fmt.Fprintf(w, version)
+}
```
Comment (BSamodien) [inline apis/trips/tripsgo/versionService.go:15]: Linter, requires New Line at the end of file

File: apis/user-java/src/main/java/io/swagger/api/VersionApi.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+@javax.annotation.Generated(value = "io.swagger.codegen.languages.SpringCodegen", date = "2018-08-03T19:26:46.543Z")
+
+@Api(value = "version", description = "the version API")
+public interface VersionApi {
+    @ApiOperation(value = "", nickname = "versionGet", notes = "Returns version for systems looking to identify the current API version or tag", response = String.class, tags={  })
+    @RequestMapping(value = "/version/user-java",
+        produces = { "text/plain" }, 
+        method = RequestMethod.GET)
+    ResponseEntity<String> versionGet();
+}
```
Comment (BSamodien) [inline apis/user-java/src/main/java/io/swagger/api/VersionApi.java:17]: Linter, requires New Line at the end of file

File: apis/user-java/src/main/java/io/swagger/api/VersionApiController.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        this.request = request;
+    }
+
+    @Override
+    public ResponseEntity<String> versionGet() {
+        String version = System.getenv("APP_VERSION");
+        return new ResponseEntity<String>(version, HttpStatus.OK);
+    }
+
+}
```
Comment (BSamodien) [inline apis/user-java/src/main/java/io/swagger/api/VersionApiController.java:31]: Linter, requires New Line at the end of file

File: apis/userprofile/Dockerfile [hunk 4]
```diff
@@ -6,6 +6,14 @@ WORKDIR  /app
...
+#    This can be specified during the docker build step by adding " --build-arg build_version=<value>"
+#    App version can be accessed via the uri path /api/version/user
+#    https://vsupalov.com/docker-build-pass-environment-variables/
+ARG build_version="userprofile default"
+ENV APP_VERSION=$build_version 
+
+
 RUN npm install
 
 CMD npm start
```
Comment (BSamodien) [inline apis/userprofile/Dockerfile:19]: Linter, requires New Line at the end of file

File: apis/userprofile/handlers/version/user.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+     * produces: 
+     * responses: 200, default
+     */
+    get: function (req, res, next) {
+      let status = 200;
+      let version  = process.env.APP_VERSION;
+      res.set('Content-Type', 'text/plain');
+      res.status(status).send(version);
+    }
+};
```
Comment (BSamodien) [inline apis/userprofile/handlers/version/user.js:20]: Linter, requires New Line at the end of file

File: apis/trips/tripsgo/versionService.go (hunk 1 above)
Comment (hattan) [reply apis/trips/tripsgo/versionService.go]: No problem, added the newline.

File: apis/user-java/src/main/java/io/swagger/api/VersionApiController.java (hunk 3 above)
Comment (hattan) [reply apis/user-java/src/main/java/io/swagger/api/VersionApiController.java]: No problem, added the newline.

File: apis/userprofile/handlers/version/user.js (hunk 5 above)
Comment (hattan) [reply apis/userprofile/handlers/version/user.js]: No problem, added the newline.

**Pull request merged**